interview-questions copy.md

```diff
@@ -0,0 +1,74 @@
+Answer the following questions.
+
+First, without external resources. Challenge yourself to answer from memory.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
+
+  Your answer: The foreign key in this example would be located in the Sightings model.  The foreign key is animal_id, because it is an attribute that refers to to the specific animal in our model Animal that it belongs to.  When dealing with foreign keys, we know that these are attributes that references a candidate key
+
+  Researched answer: A perfect example for this explanation is if you have a table for a Team would likely have member_name in it.  That member_name would likely have many People in it.  People need names, and those People in the People model are team members.  So the key is member_name because it is likely referring  to the person_name key in the model.
+
+
+
+2. Which RESTful routes must always be passed params? Why?
+
+  Your answer: RESTful routes that must always be passed params are GET and DELETE.  In order for either HTTP verb to fully function, we must pass params into it in order to specifically target the method we wish to target in order to complete the action we wish to complete.
+
+  Researched answer: Specifically, params refers to the parameters being passed to the controller via a GET or POST request.
+
+
+
+3. Name three rails generator commands. What is created by each?
+
+  Your answer: 'rails generate model arbitrary_name' creates us a model within Rails that we can use to hold our data.  'rails generate controller whatever_name' creates us a controller that holds our HTTP verbs, actions, methods etc. 'rails generate migration detailed_specific_name' creates a migration file that we can use to change data elements and columns, key names, etc. within our models.
+
+  Researched answer: Migration files allow us to influence and access models without having to worry too much on breaking our data structure or throwing off the flow.
+
+
+
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+
+method="GET"    /users         with this GET method we are trying to render a list of users, or rather, users accounts and their information
+
+method="GET"    /users/1    this GET method, or HTTP verb, targets a specific user by it's ID, here we are calling to render the first user account within our list of users.
+
+method="GET"    /users/new    this GET is calling the 'new' method in our controller that carries all the logic needed to succesfully create a new account.
+
```

Comment:
New is actually the view that allows a user to see a form where they can create a new item for the database.